modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -111,6 +111,60 @@ export function handleAction(
 
 		window.open(url);
 	}
+	else if (target === 'download') {
+		event.preventDefault();
+
+		setLoading(true);
+		let filename = '';
+
+		executeAsyncAction(url, method)
+			.then((response) => {
+				const disposition = response.headers.get('Content-Disposition');
+
+				if (disposition && disposition.indexOf('attachment') !== -1) {
+					const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
```

Comment:
Can you add docs on what these regexes do? Regex is pretty unreadable what it's trying to achieve if it needed to be modified.